### ISSUE TITLE
[feature request] MySQL Optimizer Hints

### ISSUE THREAD

**Issue by mickeyreiss** [volatiletech/sqlboiler#1320]
MySQL has support for [Optimizer Hints](https://dev.mysql.com/doc/refman/5.7/en/optimizer-hints.html), which implemented with comments that appear between `SELECT` and the column list.

For example:

```sql
SELECT /*+ MAX_EXECUTION_TIME(1000) */ * FROM users;
 ```

I would like to be able to insert optimizer hints with a query mod, maybe something like this:

```go
models.Users(
	qm.OptimizerHint("MAX_EXECUTION_TIME(%d)", 1000),
).All(ctx)
```

**Comment by mickeyreiss** [comment]
Here is some WIP for adding this feature: https://github.com/volatiletech/sqlboiler/compare/master...mickeyreiss:sqlboiler:master. It looks like it would be relatively simple to add—similar to https://github.com/volatiletech/sqlboiler/pull/872.

**Comment by stephenafamo** [comment]
Can you create a PR for this? It would be much easier to review.

**Comment by mickeyreiss** [comment]
> Can you create a PR for this? It would be much easier to review.

@stephenafamo my solution isn't fully baked yet. For example, I haven't considered databases other than MySQL. I wanted to get feedback on the feature and approach before opening a pull request. If you see any issues as is, feel free to leave a comment on the commit.

**Comment by stephenafamo** [comment]
> I haven't considered databases other than MySQL

Yes, this is an issue because SQLboiler uses the same query builder for all dialects.

Personally, I think it is fine. Because it is similar to `qm.Returning` which is not supported in MySQL but we have mods for it.
